core/math/octree.h

```diff
@@ -776,6 +776,8 @@ void Octree<T, use_pairs, AL>::_remove_element(Element *p_element) {
 				o->pairable_elements.erase(I->get().E);
 			else
 				o->elements.erase(I->get().E);
+
+			_remove_element_from_octant(p_element, o);
 		}
```

Comment:
A possible danger I could see is that elements that didn't use pairs could be deleted in the first loop above, then the octant removed, then the octant owners could be pointing to deleted data and crash.

A possible solution was to totally isolate these two cases, see the version in my PR. I'm not absolutely sure on all aspects of this, as the unpairing is a bit hard to wrap your head around.

Reply:
Yeah it could possibly happen.
But it seems that the way `_remove_element_from_octant` is implemented wouldn't allow first loop to perform any removal, at least not based on emptiness of `elements` list. And it also seems to be the only place `children` list gets cleared, so it should be relatively safe.

Looking more at it makes me think that first loop is mostly useless in case of `use_pairs` as makes `_remove_element_from_octant` just go up from octant doing nothing. 
So splitting the cases might be a good idea.